service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseManagementOrder.java

```diff
@@ -68,4 +73,23 @@ public void setNextHearingFromDynamicElement(HearingDateDynamicElement nextHeari
                 .build();
         }
     }
+
+    @JsonIgnore
+    public Map<String, Object> getCCDFields() {
+        Map<String, Object> data = new HashMap<>();
+
+        if (schedule != null) {
+            data.put(SCHEDULE.getKey(), schedule);
+        }
+
+        if (recitals != null) {
+            data.put(RECITALS.getKey(), recitals);
+        }
+
+        if (action != null) {
+            data.put(ORDER_ACTION.getKey(), action);
+        }
```

Comment:
I think you could add even if null, I think null will only mean that the case field in not populated in the UI which is fine. It also seems to be the previous behaviour

Reply:
I was getting some null pointer exceptions in some tests, so rather than changing the tests I thought it better to rework the code to get those tests passing

Reply:
I think it's better not put null values, afaik some maps implementation don't allow nulls.

I would prefer not having keys if the value is is null. @swalker125 I like you like lambdas so maybe yoou culd go with:

data.compute(ORDER_ACTION.getKey(),(k, v) -> action);

Reply:
you can't add null value to Map in java so you would never have key with null. Is compute not a bit heavy and OTT for this? We're start a new map when building this, so the values will never already be there or need to be checked etc

Reply:
It is OTT. I mean it wouldn't be performance problem for sure, just suggestion if you would like to avoid if's, but you know I am never against good old if (or for! ;))

Reply:
Btw, of course you can add null value to Map in Java! It can be used as both key and value :( 

```
        Map<String, Object> map = new HashMap<>();
        map.put("key",null);
        map.put(null, "test");

        System.out.println(map.get(null));
        System.out.println(map.get("key"));
```

This code doesn't throw any exception.